### PR TITLE
Adds Minibomb Wires

### PR DESCRIPTION
**Pull request by MrLostman**
image

image

image

## Changelog
:cl:
add: Added wires to minibombs
/:cl:

### CODE REVIEW

File: austation/code/datums/wires/explosive.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/datum/wires/explosive/minibomb
+	holder_type = /obj/item/grenade/syndieminibomb
+	randomize = TRUE	//Same behaviour since no wire actually disarms it
+
+/datum/wires/explosive/minibomb/interactable(mob/user)
+	var/obj/item/grenade/syndieminibomb/P = holder
+	if(P.open_panel)
+		return TRUE
```
Comment (TheFakeElon) [inline austation/code/datums/wires/explosive.dm:8]:
```suggestion
	return P.open_panel
```

File: austation/code/game/objects/items/grenades/syndieminibomb.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+				to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
+			if (30)
+				det_time = 50
+				to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
+			if (50)
+				det_time = 1
+				to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
+		add_fingerprint(user)
+	else
+		return
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/grenades/syndieminibomb.dm:35]:
```suggestion
```

File: austation/code/game/objects/items/grenades/syndieminibomb.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	AddComponent(/datum/component/empprotection, EMP_PROTECT_WIRES)
+
+
+/obj/item/grenade/syndieminibomb/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		open_panel = !open_panel
+		to_chat(user, "<span class='notice'>You [open_panel ? "open" : "close"] the wire panel.</span>")
+	else if(is_wire_tool(I))
+		wires.interact(user)
+	else if(I.tool_behaviour == TOOL_WIRECUTTER)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/grenades/syndieminibomb.dm:19]: don't these two contradict?

File: austation/code/game/objects/items/grenades/syndieminibomb.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/obj/item/grenade/syndieminibomb
+  var/open_panel = 0
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/grenades/syndieminibomb.dm:2]: Try to use boolean values with true/false checks

Comment (MrLostman) [reply austation/code/game/objects/items/grenades/syndieminibomb.dm:2]: this was just a copy paste of c4 wires

File: austation/code/game/objects/items/grenades/syndieminibomb.dm (hunk 3 above)
Comment (MrLostman) [reply austation/code/game/objects/items/grenades/syndieminibomb.dm]: good catch

**Comment by MrLostman** [comment]
ok everything is working nicely

File: austation/code/game/objects/items/grenades/syndieminibomb.dm (hunk 4 above)
Comment (TheFakeElon) [reply austation/code/game/objects/items/grenades/syndieminibomb.dm]: be the change I want to see

File: austation/code/game/objects/items/grenades/syndieminibomb.dm [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+			if (10)
+				det_time = 30
+				to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
+			if (30)
+				det_time = 50
+				to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
+			if (50)
+				det_time = 1
+				to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
+		add_fingerprint(user)
```
Comment (TheFakeElon) [inline austation/code/game/objects/items/grenades/syndieminibomb.dm:33]: else if neuron activation.
```suggestion
	switch(I.tool_behaviour)
		if(TOOL_SCREWDRIVER)
			open_panel = !open_panel
			to_chat(user, "<span class='notice'>You [open_panel ? "open" : "close"] the wire panel.</span>")
		if(TOOL_MULTITOOL)
			wires.interact(user)
		if(TOOL_WIRECUTTER)
			switch(det_time)
				if (1)
					det_time = 10
					to_chat(user, "<span class='notice'>You set the [name] for 1 second detonation time.</span>")
				if(10)
					det_time = 30
					to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
				if (30)
					det_time = 50
					to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
				if (50)
					det_time = 1
					to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
	add_fingerprint(user)
```

**Comment by TheFakeElon** [comment]
this is a horrible idea

**Review by TheFakeElon — APPROVED** [review]
Should be good to go after this.

File: austation/code/game/objects/items/grenades/syndieminibomb.dm [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+					to_chat(user, "<span class='notice'>You set the [name] for 1 second detonation time.</span>")
+				if(10)
+					det_time = 30
+					to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
+				if (30)
+					det_time = 50
+					to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
+				if (50)
+					det_time = 1
+					to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
```
Comment (MCterra10) [inline austation/code/game/objects/items/grenades/syndieminibomb.dm:33]: can't detonate it without a multitool?

**Pull request closed without merging**